### PR TITLE
Update runtime to nodejs@16

### PR DESCRIPTION
**Pull request by FunctionalWorm**
*Issue #, if available:* #489 

*Description of changes:*

This is [@chenrui333](https://github.com/chenrui333)'s work with its conflicts resolved .
Original PR: https://github.com/aws-actions/configure-aws-credentials/pull/413

Node 12 has an end of life on April 30, 2022.

This PR updates the default runtime to [node16](https://github.blog/changelog/2021-12-10-github-actions-github-hosted-runners-now-run-node-js-16-by-default/), rather then node12.

This is supported on all Actions Runners v2.285.0 or later.

---

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws-actions/configure-aws-credentials/blob/master/CONTRIBUTING.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by magnetikonline — APPROVED** [review]
:shipit:

**Comment by FunctionalWorm** [comment]
@peterwoodworth If you have write access to this repo, would you be able to review this for us? Thanks.

**Comment by xmunoz** [comment]
Please merge this. I'm starting to get warnings in Github Actions for depending on this workflow.

**Comment by IdanYaffe** [comment]
Would appreciate it if that could be merged.

**Comment by renperez-cpi** [comment]
any update on this? hopefully can merge this soon

**Comment by peterwoodworth** [comment]
Thanks for the contribution @FunctionalWorm.

We can't merge this at this time unfortunately. See [this comment](https://github.com/aws-actions/configure-aws-credentials/issues/489#issuecomment-1276708649) for more information on the plan to update to Node 16, and please refer to that thread for further updates and concerns about this topic. 

I will reopen this PR and merge it if it is possible for us to use this PR to perform the upgrade, however I'm pretty doubtful that will end up being possible.

Thanks again!

**Pull request closed without merging**
